pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java

```diff
@@ -123,7 +123,7 @@ public enum TransformFunctionType {
           SqlTypeFamily.CHARACTER)), "date_time_convert"),
 
   DATETRUNC("dateTrunc",
-      ReturnTypes.BIGINT_FORCE_NULLABLE,
+      ReturnTypes.BIGINT,
```

Comment:
There is a type mismatch for `BIGINT` and `BIGINT NULLABLE` during literal computation.

Reply:
That does not justify this change. `DateTrunc` can return null when input is null, and we should fix the type mismatch instead of changing the return type

Reply:
done.

Reply:
I think this was the wire issue we faced before as well.